workflow.Rmd

```diff
@@ -0,0 +1,473 @@
+    map_cb <- NA
+    company_all_cb <- NA
+    port_all_cb <- NA
+    
+    investor_name_select <- list_investors_cb[b]
+    
+    print(paste0(b, ": ", investor_name_select))
+    
+    port_raw_cb <- port_raw_all_cb %>% filter(investor_name == investor_name_select)
+    
+    port_cb <- calculate_weights(port_raw_cb, "Bonds", grouping_variables)
+    
+    port_cb <- merge_in_ald(port_cb, ald_scen_cb)
+    
+    # Portfolio weight methodology
+    port_pw_cb <- port_weight_allocation(port_cb)
+    
+    company_pw_cb <- aggregate_company(port_pw_cb)
+    
+    port_pw_cb <- aggregate_portfolio(company_pw_cb)
+    
+    # Create combined outputs
+    company_all_cb <- company_pw_cb
+    
+    port_all_cb <- port_pw_cb
+    
+    if (has_map) {
+      if (data_check(company_all_cb)) {
+        map_cb <- merge_in_geography(
+          portfolio = company_all_cb,
+          ald_raw = ald_raw_cb
+        )
+        
+        map_cb <- aggregate_map_data(map_cb)
+      }
+    }
+    
+    # Technology Share Calculation
+    if (nrow(port_all_cb) > 0) {
+      port_all_cb <- calculate_technology_share(port_all_cb)
+    }
+    
+    if (nrow(company_all_cb) > 0) {
+      company_all_cb <- calculate_technology_share(company_all_cb)
+    }
+    
+    # Scenario alignment calculations
+    port_all_cb <- calculate_scenario_alignment(port_all_cb)
+    
+    company_all_cb <- calculate_scenario_alignment(company_all_cb)
+    
+    investor_results_path <- paste0(results_path, "/", investor_name_select, "/")
+    if (!dir.exists(investor_results_path)) {
+      dir.create(investor_results_path)
+    }
+    
+    if (data_check(company_all_cb)) {
+      write_rds(company_all_cb, paste0(investor_results_path, "Bonds_results_company.rda"))
+    }
+    if (data_check(port_all_cb)) {
+      write_rds(port_all_cb, paste0(investor_results_path, "Bonds_results_portfolio.rda"))
+    }
+    if (has_map) {
+      if (data_check(map_cb)) {
+        write_rds(map_cb, paste0(investor_results_path, "Bonds_results_map.rda"))
+      }
+    }
+  }
+}
+
+
+```
+
+There are options as to where to proceed from here. From this point there are options:
+/item Run the stress testing code
+/item Print static graphics and the pdf report (older code)
+/item Print the interactive report (html) and executive summary (pdf) 
+
+This is where there is a different in workflow between the online and offline case.
+The offline workflow directs a user to the pdf code that we've been using recently. There should be little stopping a user from printing a html report, however the script to do this may need to be created. 
+
+
+#### SDA Approach
+
+The SDA approach is a methodology that is used for "Other" Sectors that do not have clear technology roadmaps, such as Steel, Cement, and Aviation but do have emission factor models. 
+
+This is just a test really to understand how this code works as we may want to incorporate and not lose this work that was done by Vincent. 
```

Comment:
I think Jackson developed Vincent's idea. Is it the same?

* https://2degreesinvesting.github.io/r2dii.analysis/articles/target-sda.html
* https://2degreesinvesting.github.io/r2dii.analysis/reference/target_sda.html